turbo/snapshotsync/freezeblocks/block_reader.go

```diff
@@ -1250,6 +1253,53 @@ func (r *BlockReader) EventsByBlock(ctx context.Context, tx kv.Tx, hash common.H
 	return result, nil
 }
 
+func (r *BlockReader) EventsById(from uint64, to time.Time, limit int) ([]*heimdall.EventRecordWithTime, error) {
```

Comment:
Would suggest to rename to "EventsByIdFromSnapshots" - the reason is that by design block reader functions will first check in the DB (that is why we pass tx in most functions) and then check in snapshots if not in the DB. Given this function directly accesses snapshots. 

Or alternatively if you prefer you can follow the same pattern of passing in a tx and implementing the logic to read from the DB if tx != nil (although it is not needed at this point in time)

Reply:
also please have a look at the `BorEventReader` interface - you may want to add this function to it - or maybe not. I'm only mentioning It for your awareness. If it is added to that interface then `RemoteBlockReader` and `RemoteBackend` will need to implement it. For our current use case it is not needed.

Reply:
Renamed function for now.